lib/ram_pwrdn/ram_pwrdn.c

```diff
@@ -3,118 +3,146 @@
  *
  * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
  */
-#include <zephyr.h>
+
+#include <hal/nrf_power.h>
 #include <logging/log.h>
+#include <sys/util.h>
+#include <zephyr.h>
 
+#include <stdint.h>
 
 LOG_MODULE_REGISTER(ram_pwrdn, CONFIG_RAM_POWERDOWN_LOG_LEVEL);
 
-#if defined CONFIG_BOARD_NRF52840DK_NRF52840 || \
-	defined CONFIG_BOARD_NRF52833DK_NRF52833
+#if !defined(CONFIG_SOC_NRF52840) && !defined(CONFIG_SOC_NRF52833)
+#error "RAM power-down module is not supported on the current platform"
+#endif
 
-#include <hal/nrf_power.h>
+#define RAM_IMAGE_END_ADDR ((uintptr_t)_image_ram_end)
+#define RAM_BANK_COUNT ARRAY_SIZE(banks)
+
+struct ram_bank {
+	uintptr_t start;
+	uint8_t section_count;
+	uint16_t section_size;
+};
+
+extern char _image_ram_end[];
+
+static const struct ram_bank banks[] = {
+	{ .start = 0x20000000UL, .section_count = 2, .section_size = 0x1000 },
+	{ .start = 0x20002000UL, .section_count = 2, .section_size = 0x1000 },
+	{ .start = 0x20004000UL, .section_count = 2, .section_size = 0x1000 },
+	{ .start = 0x20006000UL, .section_count = 2, .section_size = 0x1000 },
+	{ .start = 0x20008000UL, .section_count = 2, .section_size = 0x1000 },
+	{ .start = 0x2000A000UL, .section_count = 2, .section_size = 0x1000 },
+	{ .start = 0x2000C000UL, .section_count = 2, .section_size = 0x1000 },
+	{ .start = 0x2000E000UL, .section_count = 2, .section_size = 0x1000 },
+	{ .start = 0x20010000UL, .section_count = 6, .section_size = 0x8000 },
```

Comment:
nRF52833 has only 2 sections in RAM bank 8 but I don't see any problems with powering down RAM sections that are not present in SoC.

Reply:
Right, I can add `#ifdef`